In [1]:
import pandas as pd
import numpy as np

In [2]:
ads = pd.read_csv(r'D:\226518 dataset\PY A-03\adult Data Set.csv', skipinitialspace = True)
ads

age         workclass  fnlwgt   education  education-num  \
0       39         State-gov   77516   Bachelors             13   
1       50  Self-emp-not-inc   83311   Bachelors             13   
2       38           Private  215646     HS-grad              9   
3       53           Private  234721        11th              7   
4       28           Private  338409   Bachelors             13   
...    ...               ...     ...         ...            ...   
32556   27           Private  257302  Assoc-acdm             12   
32557   40           Private  154374     HS-grad              9   
32558   58           Private  151910     HS-grad              9   
32559   22           Private  201490     HS-grad              9   
32560   52      Self-emp-inc  287927     HS-grad              9   

           marital-status         occupation   relationship   race     sex  \
0           Never-married       Adm-clerical  Not-in-family  White    Male   
1      Married-civ-spouse    Exec-managerial        Husband  White    Male   
2                Divorced  Handlers-cleaners  Not-in-family  White    Male   
3      Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4      Married-civ-spouse     Prof-specialty           Wife  Black  Female   
...                   ...                ...            ...    ...     ...   
32556  Married-civ-spouse       Tech-support           Wife  White  Female   
32557  Married-civ-spouse  Machine-op-inspct        Husband  White    Male   
32558             Widowed       Adm-clerical      Unmarried  White  Female   
32559       Never-married       Adm-clerical      Own-child  White    Male   
32560  Married-civ-spouse    Exec-managerial           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country Salary_range  
0              2174             0              40  United-States        <=50K  
1                 0             0              13  United-States        <=50K  
2                 0             0              40  United-States        <=50K  
3                 0             0              40  United-States        <=50K  
4                 0             0              40           Cuba        <=50K  
...             ...           ...             ...            ...          ...  
32556             0             0              38  United-States        <=50K  
32557             0             0              40  United-States         >50K  
32558             0             0              40  United-States        <=50K  
32559             0             0              20  United-States        <=50K  
32560         15024             0              40  United-States         >50K  

[32561 rows x 15 columns]

#### 1.	Shape of dataset

In [3]:
ads.shape

(32561, 15)

#### 2.	Print column names

In [4]:
ads.head(0)

Empty DataFrame
Columns: [age, workclass, fnlwgt, education, education-num, marital-status, occupation, relationship, race, sex, capital-gain, capital-loss, hours-per-week, native-country, Salary_range]
Index: []

In [5]:
for col in ads.columns:
    print(col)

age
workclass
fnlwgt
education
education-num
marital-status
occupation
relationship
race
sex
capital-gain
capital-loss
hours-per-week
native-country
Salary_range


#### 3.	How many men and women (sex feature) are represented in this dataset?

In [6]:
ads['sex'].value_counts()

Male      21790
Female    10771
Name: sex, dtype: int64

#### 4.	What is the average age (age feature) of women?

In [7]:
ads[ads['sex'] == 'Female']['age'].mean()

36.85823043357163

#### 5.	What is the percentage of German citizens (native-country feature)?

In [8]:
((ads['native-country'] == 'Germany').sum() / len(ads)) * 100

0.42074874850281013

#### 6. What are the mean and standard deviation of age for those who earn more than 50K per year (salary feature) and those who earn less than 50K per year?

#### mean and standard deviation of age for those who earn more than 50K per year

In [9]:
ads[ads['Salary_range'] == '>50K']['age'].mean()

44.24984058155847

In [10]:
ads[ads['Salary_range'] == '>50K']['age'].std()

10.51902771985177

#### mean and standard deviation of age for those who earn less than 50K per year

In [11]:
ads[ads['Salary_range'] == '<=50K']['age'].mean()

36.78373786407767

In [12]:
ads[ads['Salary_range'] == '<=50K']['age'].std()

14.020088490824813

#### 7. Is it true that people who earn more than 50K have at least high school education? (education – Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters or Doctorate feature)

#### for salary more than 50K

In [13]:
ads[ads['Salary_range'] == '>50K']['education'].unique()


array(['HS-grad', 'Masters', 'Bachelors', 'Some-college', 'Assoc-voc',
       'Doctorate', 'Prof-school', 'Assoc-acdm', '7th-8th', '12th',
       '10th', '11th', '9th', '5th-6th', '1st-4th'], dtype=object)

#### for salary less than 50K

In [14]:
ads[ads['Salary_range'] == '<=50K']['education'].unique()

array(['Bachelors', 'HS-grad', '11th', 'Masters', '9th', 'Assoc-acdm',
       '7th-8th', 'Some-college', 'Assoc-voc', '5th-6th', '10th',
       '1st-4th', 'Preschool', '12th', 'Doctorate', 'Prof-school'],
      dtype=object)

#### 8. Display age statistics for each race (race feature) and each gender (sex feature). Use groupby() and describe(). Find the maximum age of men of Amer-Indian-Eskimo race.

In [15]:
ads.groupby(['race','sex'])[['age']].describe()

age                                          \
                             count       mean        std   min   25%   50%   
race               sex                                                       
Amer-Indian-Eskimo Female    119.0  37.117647  13.114991  17.0  27.0  36.0   
                   Male      192.0  37.208333  12.049563  17.0  28.0  35.0   
Asian-Pac-Islander Female    346.0  35.089595  12.300845  17.0  25.0  33.0   
                   Male      693.0  39.073593  12.883944  18.0  29.0  37.0   
Black              Female   1555.0  37.854019  12.637197  17.0  28.0  37.0   
                   Male     1569.0  37.682600  12.882612  17.0  27.0  36.0   
Other              Female    109.0  31.678899  11.631599  17.0  23.0  29.0   
                   Male      162.0  34.654321  11.355531  17.0  26.0  32.0   
White              Female   8642.0  36.811618  14.329093  17.0  25.0  35.0   
                   Male    19174.0  39.652498  13.436029  17.0  29.0  38.0   

                                        
                             75%   max  
race               sex                  
Amer-Indian-Eskimo Female  46.00  80.0  
                   Male    45.00  82.0  
Asian-Pac-Islander Female  43.75  75.0  
                   Male    46.00  90.0  
Black              Female  46.00  90.0  
                   Male    46.00  90.0  
Other              Female  39.00  74.0  
                   Male    42.00  77.0  
White              Female  46.00  90.0  
                   Male    49.00  90.0

In [16]:
ads.groupby(['race','sex'])[['age']].max()

age
race               sex        
Amer-Indian-Eskimo Female   80
                   Male     82
Asian-Pac-Islander Female   75
                   Male     90
Black              Female   90
                   Male     90
Other              Female   74
                   Male     77
White              Female   90
                   Male     90

#### the maximum age of men of Amer-Indian-Eskimo race is `82`

#### 9. Among whom is the proportion of those who earn a lot (>50K) greater: married or single men (marital-status feature)? Consider as married those who have a marital-status starting with Married (Married-civ-spouse, Married-spouse-absent or Married-AF-spouse), the rest are considered bachelors.

#### For married men

In [17]:
ads[(ads['sex'] == 'Male') & (ads['marital-status'].str.startswith('Married'))]['Salary_range'].value_counts(normalize = True)

<=50K    0.559486
>50K     0.440514
Name: Salary_range, dtype: float64

#### For single Men

In [18]:
ads[(ads['sex'] == 'Male') & ~(ads['marital-status'].str.startswith('Married'))]['Salary_range'].value_counts(normalize = True)

<=50K    0.915505
>50K     0.084495
Name: Salary_range, dtype: float64

#### `Married Men`(.44) have more proportion than `Single Men`(0.08)

#### 10. What is the maximum number of hours a person works per week (hours-per-week feature)? How many people work such a number of hours, and what is the percentage of those who earn a lot (>50K) among them?

In [19]:
max = ads['hours-per-week'].max()
max

99

#### 99 is the maximum number of hours a person works per week (hours-per-week feature).

In [20]:
l = len(ads[ads['hours-per-week'] == max])
l

85

#### 85 people work such a number of hours.

In [21]:
(ads[(ads['hours-per-week'] == max) & (ads['Salary_range'] == '>50K')].shape[0] / l) * 100

29.411764705882355

#### 29.41% is the percentage of those who earn a lot (>50K) among them.

#### 11. Count the average time of work (hours-per-week) for those who earn a little and a lot (salary) for each country (native-country). What will these be for Japan?

#### For salary range less than 50K

In [22]:
df1 = ads[ads['Salary_range'] == '<=50K']
df1.groupby('native-country')[['hours-per-week']].mean()

hours-per-week
native-country                            
?                                40.164760
Cambodia                         41.416667
Canada                           37.914634
China                            37.381818
Columbia                         38.684211
Cuba                             37.985714
Dominican-Republic               42.338235
Ecuador                          38.041667
El-Salvador                      36.030928
England                          40.483333
France                           41.058824
Germany                          39.139785
Greece                           41.809524
Guatemala                        39.360656
Haiti                            36.325000
Holand-Netherlands               40.000000
Honduras                         34.333333
Hong                             39.142857
Hungary                          31.300000
India                            38.233333
Iran                             41.440000
Ireland                          40.947368
Italy                            39.625000
Jamaica                          38.239437
Japan                            41.000000
Laos                             40.375000
Mexico                           40.003279
Nicaragua                        36.093750
Outlying-US(Guam-USVI-etc)       41.857143
Peru                             35.068966
Philippines                      38.065693
Poland                           38.166667
Portugal                         41.939394
Puerto-Rico                      38.470588
Scotland                         39.444444
South                            40.156250
Taiwan                           33.774194
Thailand                         42.866667
Trinadad&Tobago                  37.058824
United-States                    38.799127
Vietnam                          37.193548
Yugoslavia                       41.600000

#### For salary range more than 50K

In [23]:
df2 = ads[ads['Salary_range'] == '>50K']
df2.groupby('native-country')[['hours-per-week']].mean()

hours-per-week
native-country                    
?                        45.547945
Cambodia                 40.000000
Canada                   45.641026
China                    38.900000
Columbia                 50.000000
Cuba                     42.440000
Dominican-Republic       47.000000
Ecuador                  48.750000
El-Salvador              45.000000
England                  44.533333
France                   50.750000
Germany                  44.977273
Greece                   50.625000
Guatemala                36.666667
Haiti                    42.750000
Honduras                 60.000000
Hong                     45.000000
Hungary                  50.000000
India                    46.475000
Iran                     47.500000
Ireland                  48.000000
Italy                    45.400000
Jamaica                  41.100000
Japan                    47.958333
Laos                     40.000000
Mexico                   46.575758
Nicaragua                37.500000
Peru                     40.000000
Philippines              43.032787
Poland                   39.000000
Portugal                 41.500000
Puerto-Rico              39.416667
Scotland                 46.666667
South                    51.437500
Taiwan                   46.800000
Thailand                 58.333333
Trinadad&Tobago          40.000000
United-States            45.505369
Vietnam                  39.200000
Yugoslavia               49.500000

#### In Japan the salary range more than 50K 

In [24]:
j_1_s = ads[ads['Salary_range'] == '>50K']
j_1_c = ads['native-country'] == 'Japan'
j_1_s.groupby(j_1_c)[['hours-per-week']].mean()

hours-per-week
native-country                
False                45.465396
True                 47.958333

#### In Japan the salary range more than 50K is 47.958333

#### In Japan the salary range less than 50K

In [25]:
j_2_s = ads[ads['Salary_range'] == '<=50K']
j_2_c = ads['native-country'] == 'Japan'
j_2_s.groupby(j_2_c)[['hours-per-week']].mean()

hours-per-week
native-country                
False                38.836885
True                 41.000000

#### In Japan the salary range more than 50K is 41.0